flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/hybrid/tiered/storage/SubpartitionSegmentIdTracker.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.io.network.partition.hybrid.tiered.storage;
+
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.common.TieredStorageSubpartitionId;
+import org.apache.flink.runtime.io.network.partition.hybrid.tiered.tier.TierProducerAgent;
+
+/**
+ * {@link SubpartitionSegmentIdTracker} is to track segment index for each subpartition. Each {@link
+ * TierProducerAgent}'s data manager has a separate {@link SubpartitionSegmentIdTracker}.
+ */
+public interface SubpartitionSegmentIdTracker {
+
+    /**
+     * Add a new segment index belong to the subpartition.
+     *
+     * @param subpartitionId the subpartition index
+     * @param segmentId the segment index to be added
+     */
+    void addSegmentIndex(TieredStorageSubpartitionId subpartitionId, int segmentId);
+
+    /**
+     * Return whether the {@link SubpartitionSegmentIdTracker} contains the segment index of the
+     * specific subpartition.
+     *
+     * @param subpartitionId the subpartition index
+     * @param segmentId the segment index to be added
+     * @return whether the tracker contains the segment index of the subpartition
+     */
+    boolean hasCurrentSegment(TieredStorageSubpartitionId subpartitionId, int segmentId);
```

Comment:
Based on the offline discussion, the segment tracker will be introduced in the subsequential PR.

When introducing the segment tracker, this will be fixed.